src/main/java/frc/robot/robot_manager/RobotManager.java

```diff
@@ -1,18 +1,207 @@
+  public final LocalizationSubsystem localization;
+  public final VisionSubsystem vision;
+  public final ImuSubsystem imu;
+  public final IntakeSubsystem intake;
+  public final QueuerSubsystem queuer;
+
+  private final double distanceToFeedSpot = 0.0;
+  private final double distanceToSpeaker = 0.0;
+
+  public RobotManager(
+      ArmSubsystem arm,
+      ShooterSubsystem shooter,
+      LocalizationSubsystem localization,
+      VisionSubsystem vision,
+      ImuSubsystem imu,
+      IntakeSubsystem intake,
+      QueuerSubsystem queuer) {
     super(SubsystemPriority.ROBOT_MANAGER, RobotState.IDLE_NO_GP);
+    this.arm = arm;
+    this.shooter = shooter;
+    this.localization = localization;
+    this.vision = vision;
+    this.imu = imu;
+    this.intake = intake;
+    this.queuer = queuer;
   }
 
   @Override
   protected void collectInputs() {}
 
+  @Override
+  protected RobotState getNextState(RobotState currentState) {
+    switch (currentState) {
+      case SPEAKER_WAITING,
+          AMP_WAITING,
+          FEEDING_WAITING,
+          PASS_WAITING,
+          IDLE_NO_GP,
+          IDLE_WITH_GP,
+          CLIMBING_1_LINEUP,
+          CLIMBING_2_HANGING -> {}
+      case SPEAKER_SCORING -> {
+        if (!queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_NO_GP);
+        }
+      }
+      case AMP_SCORING -> {
+        if (!queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_NO_GP);
+        }
+      }
+      case FEEDING_SHOOTING -> {
+        if (!queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_NO_GP);
+        }
+      }
+      case PASS_SHOOTING -> {
+        if (!queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_NO_GP);
+        }
+      }
+      case SPEAKER_PREPARE_TO_SCORE -> {
+        if (shooter.atGoal() && arm.atGoal()) {
+          setStateFromRequest(RobotState.SPEAKER_SCORING);
+        }
+      }
+
+      case AMP_PREPARE_TO_SCORE -> {
+        if (shooter.atGoal() && arm.atGoal()) {
+          setStateFromRequest(RobotState.AMP_SCORING);
+        }
+      }
+      case FEEDING_PREPARE_TO_SHOOT -> {
+        if (shooter.atGoal() && arm.atGoal()) {
+          setStateFromRequest(RobotState.FEEDING_SHOOTING);
+        }
+      }
+      case PASS_PREPARE_TO_SHOOT -> {
+        if (shooter.atGoal() && arm.atGoal()) {
+          setStateFromRequest(RobotState.PASS_SHOOTING);
+        }
+      }
+      case UNJAM -> {}
+      case INTAKING -> {
+        if (intake.hasNote() || queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_WITH_GP);
+        }
+      }
+      case OUTTAKING -> {
+        if (!intake.hasNote() && !queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_NO_GP);
+        }
+      }
+    }
+
+    return currentState;
+  }
+
+  @Override
+  protected void afterTransition(RobotState newState) {
+    switch (newState) {
+      case SPEAKER_PREPARE_TO_SCORE, SPEAKER_SCORING, SPEAKER_WAITING -> {
```

Comment:
the `SPEAKER_SCORING` state should be treated differently, since it needs to run the queuer to actually send the note through to the shooter. same for `AMP_SCORING`, `FEEDING_SHOOTING`, `PASS_SHOOTING`